lib/jwt/jwt.go

```diff
@@ -797,3 +808,30 @@ func (k *Key) VerifyPluginToken(token string, claims PluginTokenParam) (*Claims,
 	}
 	return k.verify(token, expectedClaims)
 }
+
+// SignArbitrary creates a signed JWT with an arbitrary claims set
+func (k *Key) SignAny(claims map[string]any) (string, error) {
+	var kid string
+	var err error
+	if kid, err = KeyID(k.config.PublicKey); err != nil {
+		return "", err
+	}
+
+	var alg jose.SignatureAlgorithm
+	if alg, err = AlgorithmForPublicKey(k.config.PublicKey); err != nil {
+		return "", err
+	}
+
+	return k.sign(claims, &jose.SignerOptions{
+		ExtraHeaders: map[jose.HeaderKey]interface{}{
+			"alg": alg,
+			"kid": kid,
+		},
+	})
+}
+
+// ValidateAny parses JWT and verifies its signature. Returns *all* claims,
+// not just registered claims
+func (k *Key) ValidateAny(rawToken string) (map[string]any, error) {
+	return k.validate(rawToken)
+}
```

Comment:
related to the comment on `validate`, I feel uneasy about exporting a ValidateAny method that only validates the signature and not the actual claims